src/services/mailChimp.ts

```diff
@@ -2,38 +2,75 @@ import fetch from 'node-fetch';
 import { User } from '../schema/userType';
 import { MailChimpSecrets } from './secrets';
 
-export const newMailchimpSubscription = async (user: User, mailchimpSecret: MailChimpSecrets): Promise<void> => {
+type MailchimpResponse = {
+    subscription: string;
+    status: string;
+    detail?: string;
+};
+
+type Subscription = {
+    type: string;
+    url: string;
+};
+
+export const newMailchimpSubscription = async (
+    user: User,
+    mailchimpSecret: MailChimpSecrets,
+): Promise<MailchimpResponse[]> => {
     const { kfMailchimpListId, kfMailchimpUserName, kfMailchimpApiKey, kfDatasetSubscriptionListId } = mailchimpSecret;
     const mailChimpDataCenter = kfMailchimpApiKey.split('-')[1];
-    const buff = new Buffer(`${kfMailchimpUserName}:${kfMailchimpApiKey}`);
+    const buff = Buffer.from(`${kfMailchimpUserName}:${kfMailchimpApiKey}`);
     const b64 = buff.toString('base64');
 
-    const urls = [];
-
-    if (user.acceptedDatasetSubscriptionKfOptIn)
-        urls.push(`https://${mailChimpDataCenter}.api.mailchimp.com/3.0/lists/${kfDatasetSubscriptionListId}/members/`);
-
-    if (user.acceptedKfOptIn)
-        urls.push(`https://${mailChimpDataCenter}.api.mailchimp.com/3.0/lists/${kfMailchimpListId}/members/`);
-
-    await Promise.all(
-        urls.map((url) =>
-            fetch(url, {
-                method: 'POST',
-                headers: {
-                    Authorization: `Basic ${b64}`,
-                },
-                body: JSON.stringify({
-                    email_address: user.email,
-                    status: 'subscribed',
-                    merge_fields: {
-                        FNAME: user.firstName,
-                        LNAME: user.lastName,
+    const subscriptions: Subscription[] = [];
+
+    if (user.acceptedDatasetSubscriptionKfOptIn) {
+        subscriptions.push({
+            type: 'Dataset Subscription',
+            url: `https://${mailChimpDataCenter}.api.mailchimp.com/3.0/lists/${kfDatasetSubscriptionListId}/members/`,
+        });
+    }
+
+    if (user.acceptedKfOptIn) {
+        subscriptions.push({
+            type: 'Newsletter Subscription',
+            url: `https://${mailChimpDataCenter}.api.mailchimp.com/3.0/lists/${kfMailchimpListId}/members/`,
+        });
+    }
+
+    try {
```

Comment:
Personally, I would use a for-loop

Reply:
This was already in place and tbh we will soon-ish move to Users API, i<d rather not touch the subscription logic